Formula/j/janet.rb

```diff
@@ -17,30 +18,57 @@ class Janet < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "31581c3ce8ea833fe2a60d42ce0879093be588c37d4336910dc59f891aa40eba"
   end
 
-  depends_on "meson" => :build
-  depends_on "ninja" => :build
-
   resource "jpm" do
     url "https://github.com/janet-lang/jpm/archive/refs/tags/v1.1.0.tar.gz"
     sha256 "337c40d9b8c087b920202287b375c2962447218e8e127ce3a5a12e6e47ac6f16"
   end
 
+  def syspath
+    HOMEBREW_PREFIX/"lib/janet"
+  end
+
   def install
-    system "meson", "setup", "build", *std_meson_args
-    cd "build" do
-      system "ninja"
-      system "ninja", "install"
-    end
+    # Replace lines in the Makefile that attempt to create the `syspath`
+    # directory (which is a directory outside the sandbox).
+    inreplace "Makefile", /^.*?\bmkdir\b.*?\$\(JANET_PATH\).*?$/, "# Line removed by Homebrew formula"
+
     ENV["PREFIX"] = prefix
+    ENV["JANET_BUILD"] = "\\\"homebrew\\\""
+    ENV["JANET_PATH"] = syspath
+
+    system "make"
+    system "make", "install"
+  end
+
+  def post_install
+    mkdir_p syspath unless syspath.exist?
+
     resource("jpm").stage do
+      ENV["PREFIX"] = prefix
+      ENV["JANET_BINPATH"] = HOMEBREW_PREFIX/"bin"
+      ENV["JANET_HEADERPATH"] = HOMEBREW_PREFIX/"include/janet"
+      ENV["JANET_LIBPATH"] = HOMEBREW_PREFIX/"lib"
+      ENV["JANET_MANPATH"] = HOMEBREW_PREFIX/"share/man/man1"
+      ENV["JANET_MODPATH"] = syspath
       system bin/"janet", "bootstrap.janet"
     end
   end
 
+  def caveats
+    <<~EOS
+      When uninstalling Janet, please delete the following manually:
+      - #{HOMEBREW_PREFIX}/lib/janet
+      - #{HOMEBREW_PREFIX}/bin/jpm
+      - #{HOMEBREW_PREFIX}/share/man/man1/jpm.1
+    EOS
+  end
+
   test do
-    assert_equal "12", shell_output("#{bin}/janet -e '(print (+ 5 7))'").strip
-    assert_predicate HOMEBREW_PREFIX/"bin/jpm", :exist?, "jpm must exist"
-    assert_predicate HOMEBREW_PREFIX/"bin/jpm", :executable?, "jpm must be executable"
-    assert_match prefix.to_s, shell_output("#{bin}/jpm show-paths")
+    janet = bin/"janet"
+    jpm = HOMEBREW_PREFIX/"bin/jpm"
+    assert_equal "12", shell_output("#{janet} -e '(print (+ 5 7))'").strip
+    assert_predicate jpm, :exist?, "jpm must exist"
+    assert_predicate jpm, :executable?, "jpm must be executable"
+    assert_match Regexp.new(HOMEBREW_PREFIX/"lib/janet".to_s), shell_output("#{jpm} show-paths")
```

Comment:
why not just this?

```suggestion
    assert_match HOMEBREW_PREFIX/"lib/janet", shell_output("#{jpm} show-paths")
```

Reply:
`assert_match` appears to need something that can respond to `#=~` which Pathname apparently doesn't. I can do either:

```ruby
assert_match (HOMEBREW_PREFIX/"lib/janet").to_s, shell_output("#{jpm} show-paths")
```

or

```ruby
assert_match "lib/janet", shell_output("#{jpm} show-paths")
```

The latter doesn't test that JPM has been put in the correct place but the former possibly is more resource intensive given it converts the Pathname into a String? (Although maybe that's not too bad if the Pathname has a String backing it—not sure.)

Do you have a preference?

Reply:
Probably just the former. But please do so in a separate PR, because this one already has bottles published.

Reply:
👍

Reply:
OK, thanks. Will make a separate PR.